src/app/questionnaire-item/questionnaire-form-item-group/questionnaire-form-item-group.component.ts

```diff
@@ -29,4 +30,19 @@ export class QuestionnaireFormItemGroupComponent {
   removeGroup(index) {
     this.dispatch(removeAnswer(this.linkIdPath, index));
   }
+
+  track(index: number, paths: (LinkIdPathSegment[] | LinkIdPathSegment)[]) {
+    return pathsToString(paths);
+  }
 }
+
+const pathsToString = (paths: (LinkIdPathSegment[] | LinkIdPathSegment)[]) =>
+  paths
+    .map((path) =>
+      Array.isArray(path)
+        ? pathsToString(path)
+        : isNumber(path.index)
+        ? path.linkId + ':' + path.index
+        : path.linkId
+    )
+    .join(':');
```

Comment:
is the following a problem?
```js
pathsToString([{linkId: "a"},{linkId: "b"},{linkId: "c"}])
// "a:b:c"
pathsToString([{linkId: "a"},[{linkId: "b"},{linkId: "c"}]])
// "a:b:c"
```

Reply:
Not really, as the keys should only be unique among items of the same nesting (it is used in two different ngFor).